src/kagglehub/cache.py

```diff
@@ -37,6 +41,8 @@ def get_cached_path(handle: ResourceHandle, path: Optional[str] = None) -> str:
 def get_cached_archive_path(handle: ResourceHandle) -> str:
     if isinstance(handle, ModelHandle):
         return _get_model_archive_path(handle)
+    elif isinstance(handle, DatasetHandle):
+        return _get_dataset_path(handle)
```

Comment:
The issue is here. It should return the archive path with the archive filename.

Currently, it downloads the archive file and name it as the directory.

```
name = '/tmp/tmp2_r6k2ur/datasets/sarahjeffreson/featured-spotify-artiststracks-with-metadata/2'
```

This creates a `2` file.

then, later, you try to create a directory with the same path and it fails.

Reply:
See how we create an archive path at line #43.